rspace/src/main/scala/coop/rchain/rspace/HotStore.scala

```diff
@@ -152,21 +152,11 @@ private class InMemHotStore[F[_]: Concurrent, C, P, A, K](
   def getData(channel: C): F[Seq[Datum[A]]] =
     for {
       fromHistoryStore <- getDataFromHistoryStore(channel)
-      result <- hotStoreState.modify[Seq[Datum[A]]](state => {
-                 state.data.get(channel) match {
-                   // update with what is in historyStore and return the same
-                   case None =>
-                     (
-                       state.copy(
-                         data = state.data.updated(channel, fromHistoryStore)
-                       ),
-                       fromHistoryStore
-                     )
-                   // just return what is in hot store already
-                   case Some(data) => (state, data)
-                 }
-               })
-    } yield (result)
+      result <- hotStoreState.modify[Seq[Datum[A]]] { state =>
+                 val data = state.data.getOrElse(channel, fromHistoryStore)
+                 (state.copy(data = state.data.updated(channel, data)), data)
+               }
+    } yield result
```

Comment:
This is not quite the same, in case when data is found on the channel (in _state_) we don't need to update the state with the same data.

Reply:
Fixed with using `Option.fold` method

Reply:
Here is the example using Functor#map.
```scala
state.data
 .get(channel)
 .map((state, _))
 .getOrElse(
   state.copy(data = state.data.updated(channel, fromHistoryStore)),
   fromHistoryStore
 )
```

Reply:
@tgrospic, a very elegant solution! I didn't think of that. I updated the code